legacy/src/Charts.js

```diff
@@ -74,6 +74,7 @@ export function generateChart(chartId, chartType, chartController) {
     computed: {
       hasAnnotationPlugin() {
```

Comment:
```suggestion
      hasAnnotationPlugin() {
        return this.chartOptions?.plugins?.hasOwnProperty(ANNOTATION_PLUGIN_KEY)
```

maybe better like that?

Reply:
@dangreen `Do not access Object.prototype method 'hasOwnProperty' from target object.`

Reply:
@thabarbados 
```
Reflect.hasOwnProperty(this.chartOptions?.plugins, ANNOTATION_PLUGIN_KEY)
```

Reply:
@dangreen has the same error((

Reply:
@thabarbados 

```js
const plugins = this.chartOptions?.plugins

return plugins && Reflect.has(plugins, ANNOTATION_PLUGIN_KEY)
```

Reply:
@dangreen  has Vue error in build version`[Vue warn]: Property or method "chartOptions" is not defined on the instance but referenced during render. Make sure that this property is reactive, either in the data option, or for class-based components, by initializing the property. See: https://vuejs.org/v2/guide/reactivity.html#Declaring-Reactive-Properties.`

Reply:
@thabarbados ok. please find short variant of current code, cause current variant is too long

Reply:
@dangreen fixed